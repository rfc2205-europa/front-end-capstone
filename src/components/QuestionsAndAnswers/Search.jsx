import React from 'react';

class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      input: ''
    }
  }

  //sets input in state to values typed into form
  handleChange = (e) => {
    this.setState({
      input: e.target.value
    })
    if (this.state.input.length > 1) {
      this.handleSearch()
    }
  }

  handleSearch = () => {
    console.log(this.state.input)
    this.props.search(this.state.input, this.state.input.length)
  }


  render() {
    return (
      <div className="search" >
        <form>
          <input type="text" placeholder="Have a Question Search For Answers" onChange={this.handleChange}>
          </input>
        </form>
      </div>
    )
  }
}

export default Search;