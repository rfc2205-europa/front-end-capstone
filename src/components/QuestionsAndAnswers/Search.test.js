import renderer from 'react-test-renderer';
import React from 'react';
import Search from './Search';

describe("Search Component Testing", function () {
  it('Renders the Search component correctly', () => {
    const component = renderer.create(<Search/>);
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
  });

});