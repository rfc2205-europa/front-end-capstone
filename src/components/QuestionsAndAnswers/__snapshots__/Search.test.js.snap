// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component Testing Renders the Search component correctly 1`] = `
<div
  className="qSearchForm"
>
  <form>
    <input
      className="qSearch"
      onChange={[Function]}
      placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
      type="text"
    />
  </form>
</div>
`;
